Modules/_ctypes/cfield.c

```diff
@@ -172,10 +181,14 @@ PyCField_FromDesc(PyObject *desc, Py_ssize_t index,
         break;
 
     case EXPAND_BITFIELD:
-        *poffset += dict->size - *pfield_size/8;
-        *psize += dict->size - *pfield_size/8;
+        if (pack)
+            while(size * 8 < (*pbitofs + bitsize))
+                size += 1;
```

Comment:
Here we increment the size until it is able to hold the currently open bitfield and the new bitfield item.

Reply:
Why not:

```C
size_t target_size = *pbitofs + bitsize;
if (size * 8 < target_size) {
    size += (target_size - (size * 8)) / 8;
}
```

(Didn't work out the rounding, but I think the point stands - you can do this in a single arithmetic expression rather than a loop, no?)

In any case, it's preferable to use curly braces after control structures even for a single statement, from [PEP 7](https://www.python.org/dev/peps/pep-0007/):

> Code structure: one space between keywords like if, for and the following left paren; no spaces inside the paren; braces are required everywhere, even where C permits them to be omitted, but do not add them to code you are not otherwise modifying. All new C code requires braces. Braces should be formatted as shown:

Reply:
I guess `(*pbitofs + bitsize - 1) / 8 + 1` works.